sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -249,18 +277,25 @@ class QueryExecution(
      * Dumps debug information about query execution into the specified file.
      *
      * @param maxFields maximum number of fields converted to string representation.
+     * @param explainMode the explain mode to be used to generate the string
+     *                    representation of the plan.
      */
-    def toFile(path: String, maxFields: Int = Int.MaxValue): Unit = {
+    def toFile(path: String,
+        maxFields: Int = Int.MaxValue,
+        explainMode: ExplainMode = ExtendedMode): Unit = {
```

Comment:
I would let it take a string instead consistently by `ExplainMode.fromString(mode)` since `ExtendedMode` under `execution` is strictly private.

Reply:
@HyukjinKwon Makes sense. Thank you. I will make the change.